### ISSUE TITLE
Bug in ListMLE?

### ISSUE THREAD

**Issue by PxYu** [ptl2r/ptl2r.github.io#4]
I noticed that in the code for ListMLE:

```
def inner_train(self, batch_preds, batch_stds, **kwargs):
	# loss function of ListMLE
	batch_logcumsumexps = apply_LogCumsumExp(batch_preds)
	batch_loss = torch.sum(batch_logcumsumexps - batch_preds)
	self.optimizer.zero_grad()
	batch_loss.backward()
	self.optimizer.step()

	return batch_loss
```

`batch_stds` was not actually used... How would you calculate loss without labels?

**Comment by raymongL** [comment]
same question when reading the ListMLE paper.

**Comment by y-research-yu** [comment]
@PxYu @raymongL sorry for my late reply. Since the documents associated with the same query have been pre-sorted according to their standard labels in a decreasing order, thus batch_stds is not used here to re-order again. To some extent, it is aimed to enhance the efficiency.
